### PR TITLE
Validate certs

### PR DESCRIPTION
**Pull request by ypaq**

### CODE REVIEW

**Comment by ypaq** [comment]
(cc @fred)

**Comment by ferd** [comment]
The changes appear to make sense here. I'm a bit annoyed we have no way to pass a per-client SSL connection (the ENV is global to all of them), but that stuff would have to go on a requirements sheet for a rewrite.

File: src/gen_lockstep.erl [hunk 1]
```diff
@@ -372,17 +372,31 @@ parse_uri(Url) ->
...
 
+ssl_opts(Hostname) ->
+    case application:get_env(lockstep, verify_ssl, false) of
+        true ->
+            VerifyFun = {fun ssl_verify_hostname:verify_fun/3, [{check_hostname, Hostname}]},
+            CACerts = certifi:cacerts(),
+            [{verify, verify_peer},
+             {depth, 2},
+             {cacerts, CACerts},
+             {verify_fun, VerifyFun}];
```
Comment (ferd) [inline src/gen_lockstep.erl]: You could pass in `{server_name_indication, Hostname}` so we send SNI headers as we update from TCP to SSL.

**Comment by ypaq** [comment]
Agreed, it's annoying not able to pass a config per client. This is a less invasive change though. I'm open to the idea of a rewrite.

**Comment by ferd** [comment]
Yeah we had discussed rewrites before. This changes brings us to a safe-but-ugly state, which is nicer than our current unsafe-and-ugly. +1.

**Pull request merged**